lib/pmf_nfpse_generator.rb

```diff
@@ -219,15 +217,32 @@ def billing_date_cannot_be_in_the_future
     date_current = Date.current
     date = billing_date.try(:to_date)
     if date != date_current && date > date_current
-      errors.add(:billing_date, I18n.t("billing_date.cannot_be_in_the_future"))
+      errors.add(:billing_date, :not_be_future)
     end
   end
 
   def cpf_cnpj_format
     return unless cpf_cnpj.present?
     unless format_cpf_cnpj.size == 14 || format_cpf_cnpj.size == 11
-      errors.add(:cpf_cnpj, "Wrong CPF/CNPJ '#{cpf_cnpj}'")
+      errors.add(:cpf_cnpj, :invalid)
     end
   end
 
+  def zipcode_in_postmon_api
+    return unless zipcode.present?
+    cep = zipcode_strip(zipcode)
+    return errors.add(:zipcode, :length) unless cep.size == 8
+    response = validate_zipcode_in_postmon_api(cep)
+    return errors.add(:zipcode, :invalid) unless response.code == 200
+    parsed_response = response.parsed_response
+    return errors.add(:zipcode, :invalid) if parsed_response['cidade_info'].blank?
+  end
+
+  def zipcode_strip(cep)
+    cep.delete('.').delete('/').delete('-').delete(' ')
```

Comment:
This could be a before validation or a call at initialize... Always strip the zipcode before everything starts...